avm/res/storage/storage-account/blob-service/main.bicep

```diff
@@ -78,7 +78,7 @@ resource blobServices 'Microsoft.Storage/storageAccounts/blobServices@2022-09-01
     automaticSnapshotPolicyEnabled: automaticSnapshotPolicyEnabled
     changeFeed: changeFeedEnabled ? {
       enabled: true
-      retentionInDays: changeFeedRetentionInDays
+      retentionInDays: changeFeedRetentionInDays ?? null
```

Comment:
```suggestion
      retentionInDays: changeFeedRetentionInDays
```

Not required as `??` takes the first non-null value. Having `null` as an 'else' case means it's just ignored 😄

Reply:
The idea was to actually supply a null value. Having any value in `changeFeedRetentionInDays ` means its not stored indefinetely.

ChangefeedRetention works as follows: values from 1 to 146000 sets a specific point in time when the data gets deleted. setting `null` (not 0!) causes the data to be retained indefinitely - which is the goal here. So setting `??` should make sense, right?

Reply:
Nono absolutely. But as the parameter is nullable, not setting it results in a `null` and a `null` would be passed to the resource. 

With `retentionInDays: changeFeedRetentionInDays ?? null` all you're getting after the evaluation is `retentionInDays: null ?? null` and Bicep will pick any `null`.

Reply:
@fblix, could we move forward on this? This PR has been open for a very very long time. If bad comes to worse I can also create the contribution, yet would need to you review and approve it.